walter_nicholas/server.js

```diff
@@ -0,0 +1,28 @@
+var bodyParser = require('body-parser');
+var fs = require('fs');
+var express = require('express');
+var app = express();
+var routename = '';
+
+app.use(bodyParser.json());
+
+app.post('/:name', function(req, res) {
+  fs.writeFileSync((__dirname + '/data/' + req.params.name +'.json'), JSON.stringify(req.body));
+  routename = req.params.name;
+  console.log(req.body);
+  res.send(req.body);
+});
+
+app.get('/' + routename, function (req, res) {
```

Comment:
you can just use '/:name' here on this get post as well. And then use req.params.name